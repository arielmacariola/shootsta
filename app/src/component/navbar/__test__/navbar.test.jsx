import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import NavBar from "./../navbar";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Renders NavBar component without crashing", () => {
  const div = document.createElement("div");
  const menuList = [
    { label: "Home", linkTo: "/" },
    { label: "Upload", linkTo: "/upload" }
  ];
  ReactDOM.render(
    <BrowserRouter>
      <NavBar brandName="shootsta" menuList={menuList} />
    </BrowserRouter>,
    div
  );
});

it("Renders NavBar correctly as expected", () => {
  const menuList = [
    { label: "Home", linkTo: "/" },
    { label: "Upload", linkTo: "/upload" }
  ];
  const { getByTestId } = render(
    <BrowserRouter>
      <NavBar brandName="shootsta" menuList={menuList} />
    </BrowserRouter>
  );
  expect(getByTestId("brandName")).toHaveTextContent("shootsta");
});

it("Matches snapshot of NavBar component", () => {
  const menuList = [
    { label: "Home", linkTo: "/" },
    { label: "Upload", linkTo: "/upload" }
  ];
  const tree = renderer
    .create(
      <BrowserRouter>
        <NavBar brandName="shootsta" menuList={menuList} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
