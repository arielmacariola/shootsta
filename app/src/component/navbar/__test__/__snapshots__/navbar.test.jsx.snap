// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot of NavBar component 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
  data-testid="navigation"
>
  <a
    className="navbar-brand"
    data-testid="brandName"
    href="/"
    onClick={[Function]}
  >
    shootsta
  </a>
  <button
    aria-controls="navbarNav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler navbar-toggler-right collapsed"
    data-target="#navbarNav"
    data-toggle="collapse"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarNav"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "#27baa1",
            }
          }
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/upload"
          onClick={[Function]}
        >
          Upload
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
